gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/versions/RegionVersionVector.java

```diff
@@ -800,6 +797,70 @@ public void recordVersionForSnapshot(T member, long version, EntryEventImpl even
     }
   }
 
+  public Map<T, RegionVersionHolder<T>> getMemToVSnapshotCopy() {
+    Map<T, RegionVersionHolder<T>> m = new HashMap();
+    m.putAll(((CopyOnWriteHashMap)memberToVersionSnapshot).getInnerMap());
+    for (Map.Entry<T, RegionVersionHolder<T>> entry : m.entrySet()) {
+      m.put(entry.getKey(), entry.getValue().clone());
```

Comment:
Can some object creation be reduced here? Like first we are doing a putAll and then overwriting entries with their clone.
Like if we take the keyset of the inner map and then iterate on it and do clone and put.

Reply:
Ok. Will do.